src/apps/config/leader.lua

```diff
@@ -169,8 +169,9 @@ local function path_setter_for_grammar(grammar, path)
       assert(grammar.type == 'struct')
       local tail_id = data.normalize_id(tail_name)
       return function(config, subconfig)
-         getter(config)[tail_id] = subconfig
-         return config
+	 local newconfig = lib.deepcopy(config)
```

Comment:
What if instead we apply the name in the setup function?  I.e. it would get set here: https://github.com/Igalia/snabb/blob/lwaftr/src/apps/config/leader.lua#L410.

I don't think that the instance name is a property of the app graph, because it's not a property of any app or link.  Regarding other processes and updates, I think a global variable on core.app would be fine as it would just exist in the main process (the one that runs the leader).

I am thinking that the whole leader/follower relationship is a bit of a distraction relative to this use case -- here we should just add an `if configuration.name then app.claim_name(configuration.name)` to the `setup_fn` and we'd be all good.  WDYT?